### DISCUSSION TITLE
Getting a type error when deploying a Nextjs prisma app to Vercel

### DISCUSSION THREAD

**Discussion by ssaran2014** [prisma/prisma#21861] [Q&A] [answered]
### Question

I am reproducing the tutorial described here: https://vercel.com/guides/nextjs-prisma-postgres

The project works fine on localhost. But when I try to deploy to Vercel, after connecting to my GitHub repo, I get the following error:
<img width="893" alt="image" src="https://github.com/prisma/prisma/assets/7292306/b8a26ea5-8e62-4774-b5f0-9f21b355d486">

The GitHub repo is here: https://github.com/ssaran2014/siddharth-saran-blogr-nextjs-prisma

The one change I made is that instead of using Vercel Postgres I'm using AWS Aurora Serverless v2 (postgresql). I do not think this is the source of the error since the project works on localhost. 

I also added postinstall to my package.json in the hopes that it would solve the problem, but it didn't.
```
 {
  "name": "hello-next",
  "version": "1.0.0",
  "description": "",
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start",
    "postinstall": "prisma generate"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "dependencies": {
    "@next-auth/prisma-adapter": "^1.0.7",
    "@prisma/client": "^5.5.2",
    "eslint-config-next": "^14.0.1",
    "next": "^14.0.1",
    "next-auth": "^4.24.4",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-markdown": "8.0.0"
  },
  "devDependencies": {
    "@types/node": "17.0.14",
    "@types/react": "17.0.38",
    "prisma": "^5.5.2",
    "typescript": "4.5.5"
  }
}
```
### Expected behavior (optional)

I expect the deployment on Vercel to succeed.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("DB_URL")
  directUrl = env("DB_URL")
}

model Post {
  id        String  @id @default(cuid())
  title     String
  content   String?
  published Boolean @default(false)
  authorId  String?
  author    User?   @relation(fields: [authorId], references: [id])
}

model Account {
  id                 String  @id @default(cuid())
  userId             String  @map("user_id")
  type               String
  provider           String
  providerAccountId  String  @map("provider_account_id")
  refresh_token      String?
  access_token       String?
  expires_at         Int?
  token_type         String?
  scope              String?
  id_token           String?
  session_state      String?
  oauth_token_secret String?
  oauth_token        String?
  user               User    @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
  @@map("accounts")
}

model Session {
  id           String   @id @default(cuid())
  sessionToken String   @unique @map("session_token")
  userId       String   @map("user_id")
  expires      DateTime
  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@map("sessions")
}

model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  createdAt     DateTime  @default(now()) @map("created_at")
  updatedAt     DateTime  @updatedAt @map("updated_at")
  emailVerified DateTime? @map("email_verified")
  image         String?
  posts         Post[]
  accounts      Account[]
  sessions      Session[]

  @@map("users")
}

model VerificationToken {
  identifier String
  token      String   @unique
  expires    DateTime

  @@unique([identifier, token])
  @@map("verificationtokens")
}

```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 
- Node version is v18.15.0

- Run `prisma -v` to see your Prisma version and paste it
Prisma and @prisma/client version is 5.5.2

**Comment by ludralph** [comment, marked as answer]
Hi @ssaran2014 👋 

Thank you for raising this question.

When I ran the command  `npm run build` locally for your project, I was able to get the exact same error you shared. I then updated the version of `Typescript` in `package.json` file from `4.5.5` to `5.2.2` and the error went away. Can you do the same and let me know if it works for you?

**Reply by a19i23** [reply]
This fixed the build for me locally. Had some other issues before I could deploy but thanks!

**Reply by abdou-roa** [reply]
this fixed the issue, Thanks!

**Comment by ssaran2014** [comment]
Hi Raphael,

Your solution worked perfectly. I updated Typescript to 5.2.2. Moreover, I
updated Node to 20.9.0 as this version of Typescript needed node =>
v18.17.0. With the updates in place, the deployment was successful. Thank
you for your assistance!

**Reply by ludralph** [reply]
Thank you for adding the additional details that helped you fully resolve the issue. It's really helpful 🙌

**Reply by mrvicthor** [reply]
i just ried it and nit didn't work

**Comment by P-A-R-U-S** [comment]
@ludralph I'm getting similar error running **yarn build**
_node --version --> v22.13.1
yarn version --> v1.22.22
tsc --version --> Version 5.7.3_

```
prisma version
Environment variables loaded from .env
prisma                  : 6.3.1
@prisma/client          : 6.3.1
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v22.13.1
TypeScript              : 4.6.4
Query Engine (Node-API) : libquery-engine acc0b9dd43eb689cbd20c9470515d719db10d0b0 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli acc0b9dd43eb689cbd20c9470515d719db10d0b0 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 6.3.0-17.acc0b9dd43eb689cbd20c9470515d719db10d0b0
Default Engines Hash    : acc0b9dd43eb689cbd20c9470515d719db10d0b0
Studio                  : 0.508.0
✨  Done in 0.75s.
```
Could you help to resolve ? And clarify what this errors actually mean ?

<img width="850" alt="Screenshot 2025-03-05 at 12 20 17 PM" src="https://github.com/user-attachments/assets/8cb34be4-da14-46af-a49a-02b41f1c64f2" />
